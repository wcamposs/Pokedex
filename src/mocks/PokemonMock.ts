const PokemonMock = {
	name: 'ivysaur',
	url: 'https://pokeapi.co/api/v2/pokemon/2/',
	id: '2',
	spriteUrl:
		'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png',
	types: [
		{
			slot: 1,
			type: {
				name: 'grass',
				url: 'https://pokeapi.co/api/v2/type/12/',
			},
		},
		{
			slot: 2,
			type: {
				name: 'poison',
				url: 'https://pokeapi.co/api/v2/type/4/',
			},
		},
	],
	abilities: [
		{
			ability: {
				name: 'overgrow',
				url: 'https://pokeapi.co/api/v2/ability/65/',
			},
			is_hidden: false,
			slot: 1,
		},
		{
			ability: {
				name: 'chlorophyll',
				url: 'https://pokeapi.co/api/v2/ability/34/',
			},
			is_hidden: true,
			slot: 3,
		},
	],
	height: 10,
	weight: 130,
	stats: [
		{
			base_stat: 60,
			effort: 0,
			stat: {
				name: 'hp',
				url: 'https://pokeapi.co/api/v2/stat/1/',
			},
		},
		{
			base_stat: 62,
			effort: 0,
			stat: {
				name: 'attack',
				url: 'https://pokeapi.co/api/v2/stat/2/',
			},
		},
		{
			base_stat: 63,
			effort: 0,
			stat: {
				name: 'defense',
				url: 'https://pokeapi.co/api/v2/stat/3/',
			},
		},
		{
			base_stat: 80,
			effort: 1,
			stat: {
				name: 'special-attack',
				url: 'https://pokeapi.co/api/v2/stat/4/',
			},
		},
		{
			base_stat: 80,
			effort: 1,
			stat: {
				name: 'special-defense',
				url: 'https://pokeapi.co/api/v2/stat/5/',
			},
		},
		{
			base_stat: 60,
			effort: 0,
			stat: {
				name: 'speed',
				url: 'https://pokeapi.co/api/v2/stat/6/',
			},
		},
	],
	evolutionChain: {
		evolution_details: [],
		evolves_to: [
			{
				evolution_details: [
					{
						gender: null,
						held_item: null,
						item: null,
						known_move: null,
						known_move_type: null,
						location: null,
						min_affection: null,
						min_beauty: null,
						min_happiness: null,
						min_level: 16,
						needs_overworld_rain: false,
						party_species: null,
						party_type: null,
						relative_physical_stats: null,
						time_of_day: '',
						trade_species: null,
						trigger: {
							name: 'level-up',
							url: 'https://pokeapi.co/api/v2/evolution-trigger/1/',
						},
						turn_upside_down: false,
					},
				],
				evolves_to: [
					{
						evolution_details: [
							{
								gender: null,
								held_item: null,
								item: null,
								known_move: null,
								known_move_type: null,
								location: null,
								min_affection: null,
								min_beauty: null,
								min_happiness: null,
								min_level: 32,
								needs_overworld_rain: false,
								party_species: null,
								party_type: null,
								relative_physical_stats: null,
								time_of_day: '',
								trade_species: null,
								trigger: {
									name: 'level-up',
									url: 'https://pokeapi.co/api/v2/evolution-trigger/1/',
								},
								turn_upside_down: false,
							},
						],
						evolves_to: [],
						is_baby: false,
						species: {
							name: 'venusaur',
							url: 'https://pokeapi.co/api/v2/pokemon-species/3/',
							spriteUrl:
								'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png',
						},
					},
				],
				is_baby: false,
				species: {
					name: 'ivysaur',
					url: 'https://pokeapi.co/api/v2/pokemon-species/2/',
					spriteUrl:
						'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png',
				},
			},
		],
		is_baby: false,
		species: {
			name: 'bulbasaur',
			url: 'https://pokeapi.co/api/v2/pokemon-species/1/',
			spriteUrl:
				'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png',
		},
	},
	growthRate: {
		name: 'medium-slow',
		url: 'https://pokeapi.co/api/v2/growth-rate/4/',
	},
	habitat: {
		name: 'grassland',
		url: 'https://pokeapi.co/api/v2/pokemon-habitat/3/',
	},
	legendary: false,
	mythical: false,
};

export default PokemonMock;
