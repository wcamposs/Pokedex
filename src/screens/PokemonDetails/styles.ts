// libraries
import { StyleSheet } from 'react-native';

// js
import colors from '../../utils/colors';

const styles = StyleSheet.create({
    header: {
        height: '10%'
    },
    container: (backgroundColor: String) => ({
        flex: 1,
        backgroundColor,
    }),
    scrollViewContainer: {
        marginTop: -12,
        marginBottom: 20,
        justifyContent: 'center',
        alignItems:'center',
        
    },
    pokemonDetails: {
        paddingTop: 12,
    },
    imageContainer: {
        width: '100%',
        alignItems: 'center',
        justifyContent: 'center',
    },
    pokemonInfoContainer: {
        backgroundColor: colors.white,
        flex: 1,
        height: '100%',
        width: '100%',
        paddingTop: 30,
        borderTopLeftRadius: 30,
        borderTopRightRadius: 30,
        justifyContent: 'center',
        alignItems: 'center',
    },
    infoContainer: {
        flex: 1,
        marginVertical: 15,
        justifyContent: 'flex-start',
        alignItems: 'flex-start',
    },
    image: {
        width: 175,
        height: 175,
    },
    sectionTitle: {
        fontSize: 18,
        fontWeight: 'bold',
        marginBottom: 12,
        marginRight: 10,
        textAlign: 'center'
    },
    idText: {
        color: colors.darkGrey,
        fontSize: 24,
        fontWeight: 'bold',
        marginRight: 8,
    },
    name: {
        color: colors.white,
        fontSize: 24,
        fontWeight: 'bold'
    },
    infoName: {
        color: colors.darkBlue,
        fontSize: 16,
        alignItems: 'center',
        justifyContent: 'center',
        fontWeight: 'normal',
    },
    divider: {
        marginBottom: 10,
        width: 300,
        height: 1
    },
    typesListContainer: {
        marginTop: 10,
        marginBottom: -10,
        width: '100%',
        flexDirection: 'row',
    },
    abilityName: {
        textAlign: 'center',
        fontSize: 18,
        fontWeight: 'normal',
        marginVertical: 5
    },
    inlineDivider: {
        fontWeight: 'bold',
        fontSize: 16,
        marginHorizontal: 16
    },
    statsContainer: {
        width: '100%',
        flexDirection: 'column'
    },
    statListContainer: {
        marginHorizontal: 55
    },
    statContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between'
    },
    statName: {
        fontSize: 16,
        fontWeight: 'bold',
        marginVertical: 5
    },
    statValue: {
        fontSize: 16,
        fontWeight: 'normal',
        marginVertical: 5
    },
    centerRow: {
        flexDirection: 'row',
        alignContent: 'center',
    },
    emptyContainer: {
        height: 70,
    },
})

export default styles;