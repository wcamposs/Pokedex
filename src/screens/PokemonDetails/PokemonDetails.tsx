// libraries
import React from 'react';
import { Image, ScrollView, Text, View } from 'react-native';
import { Divider } from 'react-native-elements';

// js
import styles from './styles';
import colors from '../../utils/colors';
import backgroundColors from '../../utils/backgroundColors';

// interfaces
import { PokemonTypes } from '../../utils/interfaces';

// interfaces
import { PokemonAbilityItem } from '../../utils/interfaces';
import { PokemonStatItem } from '../../utils/interfaces';

// components
import Chip from '../../components/Chip/Chip';
import Header from '../../components/Header/Header';
import EvolutionChain from '../../components/EvolutionChain/EvolutionChain';

function PokemonDetails({ navigation, route }: any) {
	const { pokemon } = route?.params;
	const {
		abilities = [],
		height,
		id,
		name,
		spriteUrl,
		stats = [],
		types = [],
		weight,
		evolutionChain = [],
		growthRate,
		habitat,
		legendary,
		mythical,
	} = pokemon;

	const typeName = types[0].type.name;
	const backgroundColor = backgroundColors[typeName] || colors.black;

	return (
		<View style={styles.container(backgroundColor)}>
			<Header
				navigation={navigation}
				title={name}
				showBackButton
				textColor={colors.white}
			/>

			<ScrollView
				showsVerticalScrollIndicator={false}
				style={styles.pokemonDetails}
				contentContainerStyle={styles.scrollViewContainer}>
				<View style={styles.imageContainer}>
					<View style={styles.infoContainer}>
						<View style={{ flexDirection: 'row' }}>
							<Text style={styles.idText}>{`#${id}`}</Text>
							<Text style={styles.name}>
								{name.toUpperCase()}
							</Text>
						</View>
						<View style={styles.typesListContainer}>
							{types.map((pokemonType: PokemonTypes) => {
								return (
									<React.Fragment key={pokemonType.type.name}>
										<Chip name={pokemonType.type.name} />
									</React.Fragment>
								);
							})}
						</View>
					</View>

					<View>
						<Image
							source={{ uri: spriteUrl }}
							style={styles.image}
						/>
					</View>

					<View style={styles.specialPokemonTagContainer}>
						{!legendary ? null : (
							<Chip name={'legendary'} isLegendary />
						)}
						{!mythical ? null : (
							<Chip name={'mythical'} isMythical />
						)}
					</View>
				</View>

				<View style={styles.pokemonInfoContainer}>
					<View style={[styles.infoContainer, styles.centerRow]}>
						<Text style={[styles.sectionTitle, { fontSize: 16 }]}>
							HEIGHT:
						</Text>
						<Text style={styles.infoName}>
							{(height * 0.1).toFixed(2)} m
						</Text>

						<Text style={[styles.inlineDivider]}>|</Text>

						<Text style={[styles.sectionTitle, { fontSize: 16 }]}>
							WEIGHT:
						</Text>
						<Text style={styles.infoName}>
							{(weight * 0.1).toFixed(2)} kg
						</Text>
					</View>

					<Divider style={styles.divider} />

					<View style={styles.infoContainer}>
						<Text style={styles.sectionTitle}>ABILITIES</Text>

						{!abilities.length ? (
							<View style={styles.emptyContainer} />
						) : null}
						{abilities.map((item: PokemonAbilityItem) => {
							return (
								<React.Fragment key={item.ability.name}>
									<View>
										<Text style={styles.abilityName}>
											{item.ability.name.toUpperCase()}
										</Text>
									</View>
								</React.Fragment>
							);
						})}
					</View>

					<Divider style={styles.divider} />

					<View style={[styles.statsContainer, { marginBottom: 32 }]}>
						<Text style={styles.sectionTitle}>STATS</Text>

						<View style={styles.statListContainer}>
							{stats.map((item: PokemonStatItem) => {
								return (
									<React.Fragment key={item.stat.name}>
										<View style={styles.statContainer}>
											<Text style={styles.statName}>
												{item.stat.name.toUpperCase()}:{' '}
											</Text>
											<Text style={styles.statValue}>
												{item.base_stat}
											</Text>
										</View>
									</React.Fragment>
								);
							})}
						</View>
					</View>

					<Divider style={styles.divider} />

					<Text style={styles.sectionTitle}>EVOLUTIONS</Text>
					<View style={styles.pokemonEvolutionContainer}>
						<EvolutionChain
							backgroundColor={backgroundColor}
							evolutionChain={evolutionChain}
						/>
					</View>
				</View>
			</ScrollView>
		</View>
	);
}

export default PokemonDetails;
