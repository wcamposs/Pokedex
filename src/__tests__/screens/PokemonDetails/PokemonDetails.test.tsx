import React from 'react';
import renderer from 'react-test-renderer';

import PokemonMock from '../../../mocks/PokemonMock';

import PokemonDetails from '../../../screens/PokemonDetails/PokemonDetails';

jest.useFakeTimers();

describe('<Pokemon List />', () => {
	it('renders correctly', () => {
		const navigation = jest.fn();

		const route = {
			params: {
				pokemon: PokemonMock,
			},
		};

		const tree = renderer
			.create(<PokemonDetails route={route} navigation={navigation} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
