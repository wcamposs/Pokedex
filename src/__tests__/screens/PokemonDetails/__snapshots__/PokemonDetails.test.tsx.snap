// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pokemon List /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#a3ccae",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "marginTop": 20,
        "padding": 24,
        "zIndex": 100,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFF",
              "fontSize": 32,
              "fontWeight": "bold",
              "letterSpacing": 1,
              "lineHeight": 32,
              "paddingTop": 12,
              "textAlign": "center",
              "textTransform": "uppercase",
            }
          }
        >
          ivysaur
        </Text>
      </View>
      <View />
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 20,
        "marginTop": -12,
      }
    }
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "paddingTop": 12,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "flex-start",
              "marginVertical": 15,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#f5f5f5",
                  "fontSize": 24,
                  "fontWeight": "bold",
                  "marginRight": 8,
                }
              }
            >
              #2
            </Text>
            <Text
              style={
                Object {
                  "color": "#FFF",
                  "fontSize": 24,
                  "fontWeight": "bold",
                }
              }
            >
              IVYSAUR
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "marginBottom": -10,
                "marginTop": 10,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 30,
                    "height": 20,
                    "justifyContent": "center",
                    "marginHorizontal": 4,
                    "maxHeight": 20,
                    "maxWidth": 100,
                    "textAlign": "center",
                    "width": 68,
                  },
                  Object {
                    "backgroundColor": "#00bf30",
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFF",
                    "fontSize": 10,
                    "fontWeight": "bold",
                  }
                }
              >
                GRASS
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 30,
                    "height": 20,
                    "justifyContent": "center",
                    "marginHorizontal": 4,
                    "maxHeight": 20,
                    "maxWidth": 100,
                    "textAlign": "center",
                    "width": 68,
                  },
                  Object {
                    "backgroundColor": "#b216f0",
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFF",
                    "fontSize": 10,
                    "fontWeight": "bold",
                  }
                }
              >
                POISON
              </Text>
            </View>
          </View>
        </View>
        <View>
          <Image
            source={
              Object {
                "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png",
              }
            }
            style={
              Object {
                "height": 175,
                "width": 175,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 12,
              "marginTop": -12,
              "width": "100%",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFF",
            "borderTopLeftRadius": 30,
            "borderTopRightRadius": 30,
            "flex": 1,
            "height": "100%",
            "justifyContent": "center",
            "paddingVertical": 30,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "justifyContent": "flex-start",
                "marginVertical": 15,
              },
              Object {
                "alignContent": "center",
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 12,
                  "marginRight": 10,
                  "textAlign": "center",
                },
                Object {
                  "fontSize": 16,
                },
              ]
            }
          >
            HEIGHT:
          </Text>
          <Text
            style={
              Object {
                "alignItems": "center",
                "color": "#32264d",
                "fontSize": 16,
                "fontWeight": "normal",
                "justifyContent": "center",
              }
            }
          >
            1.00
             m
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginHorizontal": 16,
                },
              ]
            }
          >
            |
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 12,
                  "marginRight": 10,
                  "textAlign": "center",
                },
                Object {
                  "fontSize": 16,
                },
              ]
            }
          >
            WEIGHT:
          </Text>
          <Text
            style={
              Object {
                "alignItems": "center",
                "color": "#32264d",
                "fontSize": 16,
                "fontWeight": "normal",
                "justifyContent": "center",
              }
            }
          >
            13.00
             kg
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#bcbbc1",
              "borderBottomWidth": 0.5,
              "height": 1,
              "marginBottom": 10,
              "width": 300,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "flex-start",
              "marginVertical": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 12,
                "marginRight": 10,
                "textAlign": "center",
              }
            }
          >
            ABILITIES
          </Text>
          <View>
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "normal",
                  "marginVertical": 5,
                  "textAlign": "center",
                }
              }
            >
              OVERGROW
            </Text>
          </View>
          <View>
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "normal",
                  "marginVertical": 5,
                  "textAlign": "center",
                }
              }
            >
              CHLOROPHYLL
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#bcbbc1",
              "borderBottomWidth": 0.5,
              "height": 1,
              "marginBottom": 10,
              "width": 300,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "flexDirection": "column",
                "width": "100%",
              },
              Object {
                "marginBottom": 32,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 12,
                "marginRight": 10,
                "textAlign": "center",
              }
            }
          >
            STATS
          </Text>
          <View
            style={
              Object {
                "marginHorizontal": 55,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginVertical": 5,
                  }
                }
              >
                HP
                :
                 
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "marginVertical": 5,
                  }
                }
              >
                60
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginVertical": 5,
                  }
                }
              >
                ATTACK
                :
                 
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "marginVertical": 5,
                  }
                }
              >
                62
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginVertical": 5,
                  }
                }
              >
                DEFENSE
                :
                 
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "marginVertical": 5,
                  }
                }
              >
                63
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginVertical": 5,
                  }
                }
              >
                SPECIAL-ATTACK
                :
                 
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "marginVertical": 5,
                  }
                }
              >
                80
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginVertical": 5,
                  }
                }
              >
                SPECIAL-DEFENSE
                :
                 
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "marginVertical": 5,
                  }
                }
              >
                80
              </Text>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginVertical": 5,
                  }
                }
              >
                SPEED
                :
                 
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "fontWeight": "normal",
                    "marginVertical": 5,
                  }
                }
              >
                60
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#bcbbc1",
              "borderBottomWidth": 0.5,
              "height": 1,
              "marginBottom": 10,
              "width": 300,
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 12,
              "marginRight": 10,
              "textAlign": "center",
            }
          }
        >
          EVOLUTIONS
        </Text>
        <View
          style={
            Object {
              "flex": 1,
              "margin": 8,
              "width": "90%",
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
